File: GeneratedEntityReaderClasses/Components/ShotEffectComponent.cs
namespace GeneratedNoitaClasses;
public class ShotEffectComponent : NoitaComponentBase,  iNoitaType<ShotEffectComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaUInt condition_effect { get; set; } = new();
	public NoitaUInt condition_status { get; set; } = new();
	public NoitaString extra_modifier { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		condition_effect.Read(s);
		condition_status.Read(s);
		extra_modifier.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		condition_effect.Write(s);
		condition_status.Write(s);
		extra_modifier.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/ShotEffectComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaUInt

from NoitaStream import NoitaStream
@dataclass
class ShotEffectComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    condition_effect:NoitaUInt = field(default_factory=NoitaUInt)
    condition_status:NoitaUInt = field(default_factory=NoitaUInt)
    extra_modifier:NoitaString = field(default_factory=NoitaString)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.condition_effect.Read(s)
        self.condition_status.Read(s)
        self.extra_modifier.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.condition_effect.Write(s)
        self.condition_status.Write(s)
        self.extra_modifier.Write(s)

